crates/variables/src/provider/vault.rs

```diff
@@ -52,12 +55,24 @@ impl Provider for VaultProvider {
             Some(prefix) => format!("{}/{}", prefix, key.as_str()),
             None => key.as_str().to_string(),
         };
-        match kv2::read::<Secret>(&client, &self.mount, &path).await {
-            Ok(secret) => Ok(Some(secret.value)),
-            // Vault doesn't have this entry so pass along the chain
-            Err(ClientError::APIError { code: 404, .. }) => Ok(None),
-            // Other Vault error so bail rather than looking elsewhere
-            Err(e) => Err(e).context("Failed to check Vault for config"),
+
+        let mut retry_count = 0;
+        loop {
+            match kv2::read::<Secret>(&client, &self.mount, &path).await {
+                Ok(secret) => return Ok(Some(secret.value)),
+                // Vault doesn't have this entry so pass along the chain
+                Err(ClientError::APIError { code: 404, .. }) => return Ok(None),
+                // Other Vault error so retry, or if at retry limit bail rather than looking elsewhere
+                Err(e) => {
+                    if retry_count >= self.retry_limit {
```

Comment:
I wonder if we should only retry on 5xx errors as other errors are unlikely to be recoverable.